modules/shared/src/main/scala/org/tessellation/security/Hash.scala

```diff
@@ -52,19 +56,34 @@ object hash {
 }
 
 trait Hashable[F[_]] {
-  def hash[A <: AnyRef](data: A): Either[Throwable, Hash]
+  def hash[A: Encoder](data: A): F[Hash]
```

Comment:
Shouldn't the typeclass be typed on the object we're hashing rather than a container?
E.g. are we requiring `IO[A]` to be hashable or `A` ?

Reply:
We are requiring any `A` that has an `Encoder` to be hashable under the effect type `F[_]`.
I don't understand your concern though, can you rephrase that?

Reply:
We've texted on Slack and tl;dr; - let's follow the pattern used by `Encoder` from `io.circe`. I will reorganize the PR

Reply:
tl;dr; that pattern does not allow for dependencies and cannot be used here